### ISSUE TITLE
django-3.2 support

### ISSUE THREAD

**Issue by mgorny** [grantjenks/python-diskcache#210]
It seems that the current version (d9b9d06614a41beff9ebce9b1bc8038d5540394e) fails tests with django-3.2.4. After modifying `tox.ini` to test `django==3.2.*`, I get:

```pytb
$ tox -e py38
GLOB sdist-make: /tmp/python-diskcache/setup.py
py38 recreate: /tmp/python-diskcache/.tox/py38
py38 installdeps: django==3.2.*, pytest, pytest-cov, pytest-django, pytest-xdist
py38 inst: /tmp/python-diskcache/.tox/.tmp/package/1/diskcache-5.2.1.zip
py38 installed: asgiref==3.3.4,attrs==21.2.0,coverage==5.5,diskcache==5.2.1,Django==3.2.4,execnet==1.9.0,iniconfig==1.1.1,packaging==20.9,pluggy==0.13.1,py==1.10.0,pyparsing==2.4.7,pytest==6.2.4,pytest-cov==2.12.1,pytest-django==4.4.0,pytest-forked==1.3.0,pytest-xdist==2.2.1,pytz==2021.1,sqlparse==0.4.1,toml==0.10.2
py38 run-test-pre: PYTHONHASHSEED='632529152'
py38 run-test: commands[0] | pytest
============================================================== test session starts ===============================================================
platform linux -- Python 3.8.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
cachedir: .tox/py38/.pytest_cache
django: settings: tests.settings (from env)
rootdir: /tmp/python-diskcache, configfile: tox.ini
plugins: cov-2.12.1, django-4.4.0, xdist-2.2.1, forked-1.3.0
gw0 [244] / gw1 [244] / gw2 [244] / gw3 [244] / gw4 [244] / gw5 [244] / gw6 [244] / gw7 [244] / gw8 [244] / gw9 [244] / gw10 [244] / gw11 [244]
.........................................................F..............F................................................................. [ 56%]
..........................................................................................................                                 [100%]
==================================================================== FAILURES ====================================================================
_____________________________________________ DiskCacheTests.test_get_or_set_callable_returning_none _____________________________________________
[gw11] linux -- Python 3.8.10 /tmp/python-diskcache/.tox/py38/bin/python

self = <tests.test_djangocache.DiskCacheTests testMethod=test_get_or_set_callable_returning_none>

    def test_get_or_set_callable_returning_none(self):
        self.assertIsNone(cache.get_or_set('mykey', lambda: None))
        # Previous get_or_set() doesn't store None in the cache.
>       self.assertEqual(cache.get('mykey', 'default'), 'default')
E       AssertionError: None != 'default'

tests/test_djangocache.py:884: AssertionError
___________________________________________________________ DiskCacheTests.test_pickle ___________________________________________________________
[gw1] linux -- Python 3.8.10 /tmp/python-diskcache/.tox/py38/bin/python

self = <tests.test_djangocache.DiskCacheTests testMethod=test_pickle>

    def test_pickle(self):
        letters = 'abcde'
        cache.clear()
    
        for num, val in enumerate(letters):
            cache.set(val, num)
    
>       data = pickle.dumps(cache)
E       TypeError: cannot pickle '_thread.RLock' object

tests/test_djangocache.py:1036: TypeError
================================================================ warnings summary ================================================================
tests/test_djangocache.py::DiskCacheTests::test_cache_write_unpicklable_object
  /tmp/python-diskcache/.tox/py38/lib/python3.8/site-packages/django/middleware/cache.py:67: RemovedInDjango40Warning: Passing None for the middleware get_response argument is deprecated.
    super().__init__(get_response)

tests/test_djangocache.py::DiskCacheTests::test_cache_write_unpicklable_object
tests/test_djangocache.py::DiskCacheTests::test_cache_write_unpicklable_object
  /tmp/python-diskcache/.tox/py38/lib/python3.8/site-packages/django/middleware/cache.py:130: RemovedInDjango40Warning: Passing None for the middleware get_response argument is deprecated.
    super().__init__(get_response)

-- Docs: https://docs.pytest.org/en/stable/warnings.html

---------- coverage: platform linux, python 3.8.10-final-0 -----------
Name                       Stmts   Miss Branch BrPart  Cover   Missing
----------------------------------------------------------------------
diskcache/__init__.py         16      2      0      0    88%   60-62
diskcache/cli.py               0      0      0      0   100%
diskcache/core.py            926     18    367     22    96%   277->exit, 311->314, 332, 334->exit, 365-366, 369->372, 376->378, 406-410, 416, 452->459, 673-676, 1193->1195, 1206->1208, 1334->1329, 1334->exit, 1604->1600, 1604->exit, 1720->1716, 1720->exit, 1801, 1890->1893, 1958->1970, 1963->1970, 2450, 2457-2460
diskcache/djangocache.py     101      2     10      1    97%   9-11, 434->444
diskcache/fanout.py          217      2     34      0    99%   490, 573
diskcache/persistent.py      282      0     76      3    99%   34->37, 148->156, 1071->exit
diskcache/recipes.py         168      2     18      1    98%   105, 295
----------------------------------------------------------------------
TOTAL                       1710     26    505     27    97%

Required test coverage of 96% reached. Total coverage: 97.25%
============================================================ short test summary info =============================================================
FAILED tests/test_djangocache.py::DiskCacheTests::test_get_or_set_callable_returning_none - AssertionError: None != 'default'
FAILED tests/test_djangocache.py::DiskCacheTests::test_pickle - TypeError: cannot pickle '_thread.RLock' object
=================================================== 2 failed, 242 passed, 3 warnings in 20.82s ===================================================
ERROR: InvocationError for command /tmp/python-diskcache/.tox/py38/bin/pytest (exited with code 1)
____________________________________________________________________ summary _____________________________________________________________________
ERROR:   py38: commands failed
```

**Comment by bnavigator** [comment]
Additionally, long deprecated `django.utils.encoding.force_text()` has been removed in Django 4:

https://github.com/grantjenks/python-diskcache/blob/7bfbce63ba127d601ddfbf2838539e34cad87616/tests/test_djangocache.py#L60

**Comment by grantjenks** [comment]
Django 3.2.* support has been tested and published in diskcache v5.4.0 on PyPI.
